src/MapIterableCapableTrait.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Dhii\Iterator;
+
+use Dhii\Invocation\Exception\InvocationExceptionInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use InvalidArgumentException;
+use stdClass;
+use Traversable;
+
+trait MapIterableCapableTrait
+{
+    /**
+     * Invokes a callback for each element of the iterable.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable            $iterable The iterable to map.
+     * @param callable                              $callback
+     * @param Stringable|string|int|float|bool|null $start    The offset of the iteration, at which to start applying the callback.
+     *                                                        Iterations will still happen on all previous elements, but the callback will not be applied.
+     *                                                        Default: 0.
+     * @param Stringable|string|int|float|bool|null $count    The number  of invocations to make. Iteration will stop when this number is reached.
+     *                                                        Pass 0 (zero) to iterate until end.
+     *                                                        Default: 0.
+     * @param array|null                            $results  This will be filled with the results of the callback, in the same order, preserving keys.
+     *
+     * @throws InvalidArgumentException     If the iterable, the callback, start, or end are invalid.
+     * @throws InvocationExceptionInterface If problem during invocation of the callback.
+     */
+    protected function _mapIterable($iterable, $callback, $start = null, $count = null, array &$results = null)
+    {
+        $iterable = $this->_normalizeIterable($iterable);
+
+        if (is_null($start)) {
```

Comment:
Consider using a ternary operator for `$start` and `$end`.

```
$start = is_null($start)
    ? 0
    : $this->_normalizeInt($start);
```

These variables do not need to be normalized when defaulted to `0`. Just a nice optimization to reduce the number of useless instructions being executed. Also reduces the code as a bonus.